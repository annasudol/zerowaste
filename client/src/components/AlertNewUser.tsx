import * as React from 'react';
import { useSubscription } from '@apollo/react-hooks';
import gql from 'graphql-tag';


const NEW_USER_SUBSCRIPTION = gql`
  subscription userCreated {
    userCreated {
        name
        email
    }
}
`;

export const AlertNewUser: React.FunctionComponent = (): React.ReactElement => {
  const { data, loading } = useSubscription(NEW_USER_SUBSCRIPTION);
  console.log(data, 'data')
  return <div>hello</div>
}