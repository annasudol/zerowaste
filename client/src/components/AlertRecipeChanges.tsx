// import * as React from 'react';
// import { useSubscription } from '@apollo/react-hooks';
// import gql from 'graphql-tag';

// const GET_USER_SUBSCRIPTION = gql`
//   subscription changesInRecipe {
//     changesInRecipe {
//     id
//     title
//     ingredients
//   }
//   }
// `;


// export const AlertRecipeChanges: React.FC = (): React.ReactElement | null => {
//   const { data } = useSubscription(GET_USER_SUBSCRIPTION);

//   if (data) {
//     console.log(data, "hello")
//   }
//   return null
// }