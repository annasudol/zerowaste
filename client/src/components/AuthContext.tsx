import * as React from 'react';
import {
    isLogged,
    getAccessToken,
    isLoggedIn,
    rememberLogin as RememberLogin,
    logoutUser as LogoutUser,
} from "../userAuth";

type Context = {
    token: string | null;
    rememberLogin(newToken: string): void;

} & Record<string, any>;

const Context = React.createContext<Context>({} as any);

export const AuthProvider: React.FC = (props) => {
    const [token, setToken] = React.useState<string | null>(getAccessToken());

    const userLogged = isLoggedIn();

    const rememberLogin = React.useCallback((newToken: string) => {
        RememberLogin(newToken)
        setToken(newToken);
    }, []);

    const logoutUser = React.useCallback(() => {
        LogoutUser();
        setToken(null);
    }, []);

    return (
        <Context.Provider
            value={{
                token,
                rememberLogin,
                getAccessToken,
                logoutUser,
                userLogged
            }}
            {...props}
        />
    )
}

export const useAuth = () => React.useContext(Context);