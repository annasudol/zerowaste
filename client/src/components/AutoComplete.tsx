import * as React from 'react';
import { products } from '../assets/data/products';
import { ListType } from '../utils/types';
import Autocomplete from '@material-ui/lab/Autocomplete';
import TextField from '@material-ui/core/TextField';

const productsTitles = products.reduce((accumulator: string[], currentValue: ListType) => {
    return [...accumulator, currentValue.title]
}, []);

interface AutoCompleteProps {
    placeholder?: string
    saveList(value: string[]): void
    form?(ingredients: string[]): void
    list?: string[] | []
    error?: boolean
}

export const AutoComplete: React.FC<AutoCompleteProps> = ({ placeholder, form, saveList, list, error = false }) => {

    const handleChange = (value: string[]) => {
        if (form) {
            form(value)
        }
        saveList(value);
    }

    return (
        <Autocomplete
            multiple
            className='mt-4 mb-4'
            style={{
                border: error ? '1px solid red' : ''
            }}
            options={productsTitles}
            getOptionLabel={option => option}
            renderOption={option => (
                <span>
                    {option}
                </span>
            )}
            renderInput={params => (
                <TextField
                    {...params}
                    label={placeholder || 'Add at least one product'}
                />
            )}
            onChange={(_event, value: string[]): void => handleChange(value)}
            value={list}
        />
    )

}