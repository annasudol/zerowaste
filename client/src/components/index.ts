export * from './App';
export * from './Button';
export * from './ErrorMessage';
export * from './Image';
export * from './LoadingBar';
export * from './LoginSignUp';
export * from './Navigation';
export * from './RecipesList';
export * from './RecipeDetails';
export * from './RecipeForm';
export * from './RecipeItem';
export * from './SearchRecipesPage';
export * from './SearchRecipesForm';