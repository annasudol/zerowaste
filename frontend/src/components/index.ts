export * from './app/App';
export * from './loginSignUpForm/LoginSignUpForm';
export * from './autoCompleteForm/AutoCompleteForm';
export * from './dialogDeleteRecipe/DialogDeleteRecipe';
export * from './errorMessage/ErrorMessage';
export * from './image/Image';
export * from './imageUpload/ImageUpload';
export * from './list/List';
export * from './protectedRoute/ProtectedRoute';
export * from './loadingBar/LoadingBar';
export * from './listInput/ListInput';
export * from './navigation/Navigation';
export * from './recipeItem/RecipeItem';
export * from './recipeForm/RecipeForm';
export * from './searchRecipesForm/SearchRecipesForm';