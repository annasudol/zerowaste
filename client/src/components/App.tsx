import * as React from "react";
import { Navigation } from "../components";
import { Login, SignUp, Main, RecipesList, AddRecipeForm, RecipeInfo, UserInfo } from "../pages";

import { AppRoutes } from '../../routes';
import { Route, Switch } from 'react-router-dom';
import { ErrorMessage } from './ErrorMessage';


export const App: React.FunctionComponent = () => {
    const token = window.localStorage.getItem('token');

    return (
        <>
            <Navigation token={token} />
            <Switch>
                <Route path={AppRoutes.Home} component={Main} exact />
                <Route path={AppRoutes.RecipesList} component={RecipesList} />
                <Route path={AppRoutes.Recipe} component={RecipeInfo} />

                {!token ? (
                    <>
                        <Route path={AppRoutes.Login} component={Login} />
                        <Route path={AppRoutes.SignUp} component={SignUp} />
                    </>
                ) : (
                        <>
                            <Route path={AppRoutes.User} component={UserInfo} />
                            <Route path={AppRoutes.AddRecipe} component={AddRecipeForm} />
                        </>
                    )}
                <Route path="*">
                    <ErrorMessage message="page not found" />
                </Route>
            </Switch>
        </>
    )
}