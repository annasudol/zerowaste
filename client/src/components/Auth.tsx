import * as React from "react";
import { OnSubmit } from "react-hook-form";
import { LoadingBar, ErrorMessage } from "../components";
import { useHistory } from 'react-router';
import { AppRoutes } from "../../routes";
type Inputs = {
    name?: string
    email: string
    password: string
}

interface AuthProps {
    //    loading: boolea n
    errorMessage?: string
    register: any
    handleSubmit: (callback: OnSubmit<Inputs>) => (e?: React.BaseSyntheticEvent<object, any, any> | undefined) => Promise<void>
    submit: any
    loginPage?: boolean
}

export const Auth: React.FunctionComponent<AuthProps> = ({ errorMessage, handleSubmit, register, submit, loginPage = true }): React.ReactElement => {
    const history = useHistory();

    // if (data) {
    //     return history.push({ pathname: `${AppRoutes.Home}` });
    // }
    return (
        <div className="content overflow-hidden flex justify-center items-center">
            <div className="form  flex flex-col">
                <h1 className="form-header font-bebas uppercase text-darkGray text-center pb-0 m-0">{loginPage ? 'Log in' : 'Sign Up'}</h1>
                {errorMessage && <p className="text-coral text-sm text-center mb-3">{errorMessage}</p>}
                <form onSubmit={handleSubmit(submit)}>
                    {!loginPage && <input name="name" type="text" placeholder="Name" ref={register({ required: true, maxLength: 30 })} />}
                    <input
                        name="email"
                        placeholder="Email"
                        type="email"
                        ref={register({
                            required: true,
                            pattern: {
                                value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
                                message: "invalid email address"
                            }
                        })}
                    />
                    <input name="password" type="password" placeholder="Set a password" ref={register({ required: true })} />
                    <input type="submit" title="Submit" className="bg-green hover:bg-green text-white font-bold py-2 px-4 rounded outline-none m-1 border-0" />
                </form>
            </div>
        </div >)
}
