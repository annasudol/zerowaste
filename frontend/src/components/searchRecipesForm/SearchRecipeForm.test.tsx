import React from 'react';
import { shallow } from 'enzyme';
import { SearchRecipesForm, AutoCompleteForm } from '../../components';
import toJson from 'enzyme-to-json';
import * as redux from 'react-redux';
import { Button } from 'antd';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
    useDispatch: (): jest.Mock<any, any> => mockDispatch,
    useSelector: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
    useHistory: () => ({
        push: jest.fn(),
    }),
}));


jest.mock('react-redux', () => ({
    useDispatch: (): jest.Mock<any, any> => mockDispatch,
    useSelector: jest.fn(),
}));


describe('component', (): void => {
    describe('SearchRecipesForm', (): void => {
      it('renders component when getProducts is undefined', (): void => {
        const wrapper = shallow(<SearchRecipesForm />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });

      it('renders component when getProducts is given', (): void => {
        const spy = jest.spyOn(redux, 'useSelector');

        spy.mockReturnValue({getProducts: ["apple", "sugar"]});
        });
        const wrapper = shallow(<SearchRecipesForm />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('calls dispatch function after clicking the button', (): void => {
        const spy = jest.spyOn(redux, 'useSelector');

        spy.mockReturnValue({getProducts: ["apple", "sugar"]});

        const wrapper = shallow(<SearchRecipesForm />);
        wrapper.find(AutoCompleteForm).prop('saveList')(['apple']);
        wrapper.find(Button).simulate('click');
        expect(mockDispatch).toHaveBeenCalledTimes(1);


    });
  });
  