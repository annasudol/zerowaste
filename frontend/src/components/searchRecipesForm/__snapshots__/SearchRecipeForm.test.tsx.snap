// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Fragment>
  <AutoCompleteForm
    saveList={[Function]}
  />
</Fragment>
`;

exports[`component SearchRecipesForm renders component when getProducts is undefined 1`] = `
<Fragment>
  <AutoCompleteForm
    saveList={[Function]}
  />
</Fragment>
`;
